### PR TITLE
`@turf/concave` refactoring

### PR DESCRIPTION
**Pull request by stebogit**
I was working to fixing #333, but it seems it's not actually an issue anymore.
The [original jsfiddle](https://jsfiddle.net/rnisthal/7tsygLrs/) was using to Turf v2.0.
Now, at least after the refactoring, the error does not occur.
@DenisCarriere the [case](https://github.com/Turfjs/turf/issues/333#issuecomment-246165916) you added currently does not calculate a polygon because `maxEdge` is too small; so it now errors out with that message.

While refactoring the module I noticed two things:
- `@turf/concave` is currently using `@turf/union`, thus it means it's based on `jsts` 👎 
- #310, mentioned in #333, basically suggests `@turf/concave` should be now implemented with [`concaveman`](https://github.com/mapbox/concaveman) (👍 ), but it's not. It seems the original [`turf-concave`](https://github.com/turf-junkyard/turf-concave) is/was though, and I didn't get why the current module does not reflect the latest implementation of `turf-concave`.

Should we use `concaveman` then? I'd be happy to implement that.

cc @tmcw

### CODE REVIEW

**Comment by DenisCarriere** [comment]
> @DenisCarriere the case you added currently does not calculate a polygon because maxEdge is too small; so it now errors out with that message.

Great! Good to know it now throws an Error that makes sense.

> Should we use concaveman then? I'd be happy to implement that.

💯%! Most definitely should!

File: packages/turf-concave/index.js [hunk 1]
```diff
@@ -31,40 +32,41 @@ var distance = require('@turf/distance');
...
+function merge(features) {
+    var merged = clone(features[0]);
+    merged.properties = {};
 
     for (var i = 0, len = features.length; i < len; i++) {
         var poly = features[i];
-        if (poly.geometry) {
-            merged = union(merged, poly);
-        }
+        if (poly.geometry) merged = union(merged, poly);
```
Comment (DenisCarriere) [inline packages/turf-concave/index.js]: Just an FYI, we are doing two `clone` operation here, one at the very top (which is fine) and a 2nd time internally with `union`. No need to change this, but we should one day revamp `union` (can be discussed separately).

**Comment by stebogit** [comment]
Ok @DenisCarriere I'll implement `@turf/concave` with `concaveman`. Should I create a new PR for that?
I'd keep working on this branch however.

**Comment by DenisCarriere** [comment]
@stebogit 👍 any branch is good

**Pull request merged**